src/platform_impl/windows/status_bar.rs

```diff
@@ -0,0 +1,304 @@
+            }
+          }
+        }
+      }
+    }
+  }
+
+  // set the icon for our main instance
+  fn set_icon(&self, icon: HICON) {
+    unsafe {
+      let mut nid = get_nid_struct(&self.hwnd);
+      nid.uFlags = NIF_ICON;
+      nid.hIcon = icon;
+      if shellapi::Shell_NotifyIconW(NIM_MODIFY, &mut nid as *mut NOTIFYICONDATAW) == 0 {
+        debug!("Error setting icon");
+        return;
+      }
+    }
+  }
+
+  pub fn shutdown(&self) {
+    unsafe {
+      let mut nid = get_nid_struct(&self.hwnd);
+      nid.uFlags = NIF_ICON;
+      if shellapi::Shell_NotifyIconW(NIM_DELETE, &mut nid as *mut NOTIFYICONDATAW) == 0 {
+        debug!("Error removing icon");
+        return;
+      }
+    }
+  }
+}
+// basic NID for our icon
+pub(crate) fn get_nid_struct(hwnd: &HWND) -> NOTIFYICONDATAW {
+  NOTIFYICONDATAW {
+    cbSize: std::mem::size_of::<NOTIFYICONDATAW>() as DWORD,
+    hWnd: *hwnd,
+    uID: 0x1 as UINT,
+    uFlags: 0 as UINT,
+    uCallbackMessage: 0 as UINT,
+    hIcon: 0 as HICON,
+    szTip: [0 as u16; 128],
+    dwState: 0 as DWORD,
+    dwStateMask: 0 as DWORD,
+    szInfo: [0 as u16; 256],
+    u: Default::default(),
+    szInfoTitle: [0 as u16; 64],
+    dwInfoFlags: 0 as UINT,
+    guidItem: GUID {
+      Data1: 0 as c_ulong,
+      Data2: 0 as c_ushort,
+      Data3: 0 as c_ushort,
+      Data4: [0; 8],
+    },
+    hBalloonIcon: 0 as HICON,
+  }
+}
+
+unsafe extern "system" fn subclass_proc(
+  h_wnd: HWND,
+  msg: UINT,
+  w_param: WPARAM,
+  l_param: LPARAM,
+) -> LRESULT {
+  if msg == winuser::WM_MENUCOMMAND {
+    WININFO_STASH.with(|stash| {
+      let stash = stash.borrow();
+      let stash = stash.as_ref();
+      if let Some(stash) = stash {
+        let menu_id = winuser::GetMenuItemID(stash.status_bar.hmenu, w_param as i32) as u32;
+        stash.handler.send_click_event(menu_id, MenuType::Statusbar);
+      }
+    });
+  }
+
+  if msg == winuser::WM_DESTROY {
+    winuser::PostQuitMessage(0);
+  }
+
+  // track the click
+  if msg == WM_USER + 1 {
+    if l_param as UINT == winuser::WM_LBUTTONUP || l_param as UINT == winuser::WM_RBUTTONUP {
+      let mut p = POINT { x: 0, y: 0 };
+      if winuser::GetCursorPos(&mut p as *mut POINT) == 0 {
+        return 1;
+      }
+      // set the popup foreground
+      winuser::SetForegroundWindow(h_wnd);
+      WININFO_STASH.with(|stash| {
+        let stash = stash.borrow();
+        let stash = stash.as_ref();
+        if let Some(stash) = stash {
+          // track the click
+          winuser::TrackPopupMenu(
+            stash.status_bar.hmenu,
+            0,
+            p.x,
+            p.y,
+            // align bottom / right, maybe we could expose this later..
+            (winuser::TPM_BOTTOMALIGN | winuser::TPM_LEFTALIGN) as i32,
+            h_wnd,
+            std::ptr::null_mut(),
+          );
+        }
+      });
+    }
+  }
+
+  return winuser::DefWindowProcW(h_wnd, msg, w_param, l_param);
+}
+
+// FIXME: when doing hard kill it doesnt work
```

Comment:
Tested with normal `ControlFlow::Exit` and it can be closed properly.
We don't need to handle this since hard kill is considered exception imho.